src/client/src/rt-platforms/glue/window.ts

```diff
@@ -0,0 +1,182 @@
+
+let listOfOpenedWindows: GDWindow[] = []
+
+// TODO are there better ways to check?
+const isSpot = (url: string) => url.indexOf('spot') >= 0
+const isAnalytics = (url: string) => url.indexOf('analytics') >= 0
+
+/**
+ * Glue42 has the ability to manage windows. The registered methods here give the ability to demonstrate different
+ * functionalities with the opened windows (Activity, Blotter, Spot windows).
+ */
+export const registerWindowMethods = () => {
+  if (isSpot(location.href)) {
+    return
+  }
+
+  let isCollapsed = false
+  window.glue.interop.register('toggleCollapse', async () => {
+    for (const wnd of listOfOpenedWindows) {
+      isCollapsed ? await wnd.expand() : await wnd.collapse()
+    }
+    isCollapsed = !isCollapsed
+  })
+
+  window.glue.interop.register('stackAllWindows', () => {
+    listOfOpenedWindows.forEach(async (wnd, index) => {
+      if (index > 0) {
+        await wnd.detachTab({
+          relativeTo: listOfOpenedWindows[index - 1],
+          relativeDirection: 'bottom',
+        })
+      }
+    })
+  })
+
+  window.glue.interop.register('tabAllWindows', () => {
+    listOfOpenedWindows.forEach(async (wnd, index) => {
+      if (index > 0) {
+        await listOfOpenedWindows[0].attachTab(wnd, index)
+      }
+    })
+  })
+
+  window.glue.interop.register('openWorkspace', (args: { symbol: string }) => {
+    // canvas typings are currently external
+    ;(window.glue42gd as GDObject & { canvas: CanvasAPI }).canvas.openWorkspace(
+      'Reactive Trader Workspace',
+      { context: args },
+    )
+  })
+
+  window.glue.windows.onWindowRemoved((removedWnd: GDWindow) => {
+    listOfOpenedWindows = listOfOpenedWindows.filter(wnd => wnd.id !== removedWnd.id)
+    window.glue.interop.invoke('toggleHeaderButtons', {
+      numberOfOpenedWindows: listOfOpenedWindows.length,
+    })
+  })
+}
+
+export const openGlueWindow = async (config: BrowserWindowProps, onClose?: () => void) => {
+  const myWindow: GDWindow = window.glue.windows.my()
+  const { name, width, height, url } = config
+  const {
+    left,
+    top,
+    modifiedWidth,
+    modifiedHeight,
+    relativeTo,
+    relativeDirection,
+  } = calculatePosition(myWindow, width, height, url)
+  const fullUrl = `${location.href.slice(0, -1)}${url}`
+  const isTabWindow = isSpot(url)
+
+  const win = await window.glue.windows.open(name, fullUrl, {
+    title: capitalizeFirstLetter(name.split(' ')[0]),
+    width: Math.round(modifiedWidth),
+    height: Math.round(modifiedHeight),
+    left: Math.round(left),
+    top: Math.round(top),
+    relativeTo,
+    relativeDirection,
+    allowCollapse: false,
+    mode: 'tab',
+    tabGroupId: isTabWindow ? 'reactiveTraderCloudSpot' : name,
+  })
+
+  if (isTabWindow) {
+    listOfOpenedWindows.push(win)
+    window.glue.interop.invoke('toggleHeaderButtons', {
+      numberOfOpenedWindows: listOfOpenedWindows.length,
+    })
+  }
+
+  if (win) {
+    if (onClose) {
+      win.onClose(onClose)
```

Comment:
Should that not be `win.onClose = onClose` instead?

Reply:
@fdore I tried setting `win.onClose = onClose`, it won't reset the tiles back to parent on close. This isn't a browser, it's Glue42

Reply:
👍  should have checked the api